### DISCUSSION TITLE
a question about  json_parser

### DISCUSSION THREAD

**Discussion by loveyang2012** [vectordotdev/vector#17817] [Q&A] [answered]
### Discussed in https://github.com/vectordotdev/vector/discussions/17801

<div type='discussions-op-text'>

<sup>Originally posted by **loveyang2012** June 29, 2023</sup>
I have gotten message like this:
{"log":"this is 456","tag":"demotag1","timestamp":"2023-06-29T22:08:58.758192934Z"}
{"log":"this is 123","tag":"demotag2","timestamp":"2023-06-29T22:08:58.580667480Z"}

when json_parser transforms  is deprecated,how can I only get message like "this is 456" and  "this is 123",yes, only get json's value ,
and  I want sent "this is 456" to kafka1 ,"this is 123" to kafka2.  may i get a  example,please</div>

**Comment by jszwedko** [comment, marked as answer]
Hi @loveyang2012 ! Just a note that I converted this back into a discussion since it doesn't seem like a bug report. You can use the `route` transform to route events based on content.

**Reply by loveyang2012** [reply]
thinks!
the config like this

```
[sources.a_coming]
type = "fluent"
address = "0.0.0.0:8225"
receive_buffer_bytes = 65_536

[transforms.b]
inputs = ["a_coming"]
type = "route"
[transforms.b.route]
r1 = '.tag == "de.tag"'

[sinks.out]
inputs = ["b.r1"]
type = "console"
encoding.codec = "json"
```

The original message is "345", and I get messges by vector like this:

{"host":"172.17.0.2","log":"345","source_type":"fluent","tag":"de.tag","timestamp":"2023-07-01T10:27:02.085643247Z"}
{"host":"172.17.0.2","log":"345","source_type":"fluent","tag":"de.tag","timestamp":"2023-07-01T10:26:46.027223259Z"}
{"host":"172.17.0.2","log":"345","source_type":"fluent","tag":"de.tag","timestamp":"2023-07-01T10:27:03.087210640Z"}

But this is not I want , I just want  message like "345" only ,not the whole line, can I get the original messages?

**Reply by jszwedko** [reply]
You can try setting `encoding.codec = "text"`. This will only print out the `message` field from the event. In your case, you will need to remap `log` to `message` though.
